assembl/static2/js/app/pages/idea.jsx

```diff
@@ -9,38 +9,62 @@ import IdeaWithPosts from '../graphql/IdeaWithPosts.graphql';
 import InfiniteSeparator from '../components/common/infiniteSeparator';
 import Post, { PostFolded } from '../components/debate/thread/post';
 import GoUp from '../components/common/goUp';
-import Tree, { Child } from '../components/common/tree';
+import Tree from '../components/common/tree';
 import withLoadingIndicator from '../components/common/withLoadingIndicator';
 
 import TopPostForm from './../components/debate/thread/topPostForm';
 
-export const transformPosts = (posts) => {
+export const transformPosts = (edges, additionnalProps) => {
   const postsByParent = {};
-  posts.forEach((p) => {
+  edges.forEach((e) => {
+    const p = { ...e.node, ...additionnalProps };
     const items = postsByParent[p.parentId] || [];
     postsByParent[p.parentId] = items;
     items.push(p);
   });
 
   const getChildren = (id) => {
     return (postsByParent[id] || []).map((post) => {
-      return { ...post, children: getChildren(post.id) };
+      const newPost = post;
+      // We modify the object in place, we are sure it's already a copy from
+      // the forEach edges above.
+      newPost.children = getChildren(post.id);
+      return newPost;
     });
   };
 
   return (postsByParent.null || []).map((p) => {
-    return { ...p, children: getChildren(p.id) };
+    const newPost = p;
+    newPost.children = getChildren(p.id);
+    return newPost;
   });
 };
 
+const noRowsRenderer = () => {
+  return (
+    <div className="center">
+      <Translate value="debate.thread.noPostsInThread" />
+    </div>
+  );
+};
+
 class Idea extends React.Component {
+  componentWillMount() {
+    this.refetchIdea = this.props.data.refetch;
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // data.refetchIdea is a new instance each time Idea component is rerendered
+    // so shallowEqual fails on Post components and we have a perf issue.
```

Comment:
Ok, deleted in https://github.com/assembl/assembl/commit/1c67ae082f66ba174242b6eed109f9b9254b0ee7